File: src/main/resources/helloWorldClient.py
import sys
import grpc
from resources import hello_pb2
from resources import hello_pb2_grpc

def main():
    firstname = sys.argv[1] if len(sys.argv) > 1 else "Julian"
    lastname = sys.argv[2] if len(sys.argv) > 2 else "Glueck"

    # Connect to Java gRPC server
    with grpc.insecure_channel("localhost:50051") as channel:
        stub = hello_pb2_grpc.HelloWorldServiceStub(channel)

        # --- Call hello() RPC ---
        hello_request = hello_pb2.HelloRequest(firstname=firstname, lastname=lastname)
        hello_response = stub.hello(hello_request)
        print("Hello RPC response:", hello_response.text)

        # --- Call sendWarehouse() RPC ---
        product1 = hello_pb2.ProductData(
            productId=101,
            productName="MacBook Air",
            productCategory="Electronics",
            quantity=50,
            price=999
        )
        product2 = hello_pb2.ProductData(
            productId=102,
            productName="Analog Clock",
            productCategory="Gadgets",
            quantity=30,
            price=49.99
        )

        warehouse_record = hello_pb2.WarehouseRecord(
            warehouseID="WH-001",
            warehouseName="Main Warehouse",
            timestamp="2025-12-02T10:15:30Z",
            productDataList=[product1, product2]
        )

        warehouse_response = stub.sendWarehouse(warehouse_record)
        print("\nWarehouse RPC response:", warehouse_response.text)

        print("Sent Warehouse Record:")
        print(f"ID: {warehouse_record.warehouseID}")
        print(f"Name: {warehouse_record.warehouseName}")
        print(f"Timestamp: {warehouse_record.timestamp}")
        print("Products:")
        for p in warehouse_record.productDataList:
            print(f"- {p.productId}: {p.productName} ({p.productCategory}), Quantity: {p.quantity}, Price: {p.price:.2f}")


if __name__ == "__main__":
    main()

File: src/main/java/HelloWorldClient.java
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import hello.Hello;
import hello.HelloWorldServiceGrpc;


import java.util.Arrays;

public class HelloWorldClient {

    public static void main(String[] args) {

        ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", 50051)
                .usePlaintext()
                .build();

        HelloWorldServiceGrpc.HelloWorldServiceBlockingStub stub =
                HelloWorldServiceGrpc.newBlockingStub(channel);

        // --- Call hello() RPC ---
        String firstname = args.length > 0 ? args[0] : "Julian";
        String lastname = args.length > 1 ? args[1] : "Glueck";

        Hello.HelloRequest helloRequest = Hello.HelloRequest.newBuilder()
                .setFirstname(firstname)
                .setLastname(lastname)
                .build();

        Hello.HelloResponse helloResponse = stub.hello(helloRequest);
        System.out.println("Hello Response: " + helloResponse.getText());

        // --- Call sendWarehouse() RPC ---
        Hello.ProductData product1 = Hello.ProductData.newBuilder()
                .setProductId(101)
                .setProductName("MacBook Pro")
                .setProductCategory("Electronics")
                .setQuantity(50)
                .setPrice(19.99)
                .build();

        Hello.ProductData product2 = Hello.ProductData.newBuilder()
                .setProductId(102)
                .setProductName("Digital Clock")
                .setProductCategory("Gadgets")
                .setQuantity(30)
                .setPrice(49.99)
                .build();

        Hello.WarehouseRecord warehouseRecord = Hello.WarehouseRecord.newBuilder()
                .setWarehouseID("WH-001")
                .setWarehouseName("Main Warehouse")
                .setTimestamp("2025-12-02T10:15:30Z")
                .addAllProductDataList(Arrays.asList(product1, product2))
                .build();

        Hello.HelloResponse warehouseResponse = stub.sendWarehouse(warehouseRecord);

        System.out.println("\nWarehouse Response: " + warehouseResponse.getText());
        System.out.println("Sent Warehouse Record:");
        System.out.println("ID: " + warehouseRecord.getWarehouseID());
        System.out.println("Name: " + warehouseRecord.getWarehouseName());
        System.out.println("Timestamp: " + warehouseRecord.getTimestamp());
        System.out.println("Products:");
        for (Hello.ProductData p : warehouseRecord.getProductDataListList()) {
            System.out.printf("- %d: %s (%s), Quantity: %d, Price: %.2f%n",
                    p.getProductId(),
                    p.getProductName(),
                    p.getProductCategory(),
                    p.getQuantity(),
                    p.getPrice());
        }

        channel.shutdown();
    }
}
